src/main/kotlin/com/hibob/academy/dao/PetDao.kt

```diff
@@ -24,14 +23,15 @@ class PetDao(private val sql: DSLContext) {
         )
     }
 
-    fun insertPet(name: String, type: PetType, companyId: Long, dateOfArrival: LocalDate, ownerId: Long? = null) {
-        sql.insertInto(petTable)
-            .set(petTable.name, name)
-            .set(petTable.type, type.toString())
-            .set(petTable.companyId, companyId)
-            .set(petTable.dateOfArrival, dateOfArrival)
-            .set(petTable.ownerId, ownerId)
-            .execute()
+    fun insertPet(petWithoutId: PetWithoutId): Long? {
```

Comment:
We prefer not to return nullable id - if it is null - then we have and issue and maybe it would be better to throw an exception this case
If it is not null - just return the is as Long and not Long?

Reply:
got it, fixed